seedcase_sprout/core/properties.py

```diff
@@ -21,12 +21,19 @@ def default(cls: type[Self]) -> Self:
 
     @property
     def compact_dict(self) -> dict:
-        """Convert the object to a dictionary, removing any keys with None values.
+        """Converts the object to a dictionary, removing any keys with None values.
+
+        Applies recursively to nested `*Properties` objects.
 
         Returns:
             A dictionary representation of the object with only non-None values.
         """
-        return {key: value for key, value in asdict(self).items() if value is not None}
+        return asdict(
+            obj=self,
+            dict_factory=lambda tuples: dict(
+                tuple for tuple in tuples if tuple[1] is not None
+            ),
+        )
```

Comment:
Would this be a bit cleaner, so we avoid the `tuple[1]`? 
```suggestion
        return asdict(
            obj=self,
            dict_factory=lambda tuples: {
                key: value for key, value in tuples if value is not None
            },
        )
```